src/node/heartbeat/sync_mempool.rs

```diff
@@ -31,20 +31,66 @@ pub async fn sync_mempool<B: Blockchain>(
             .into_iter()
             .map(|(_, r)| (r.chain_sourced, r.mpn_sourced))
             .collect::<Vec<_>>();
+        let mut fee_skipped_txs = 0;
+        let mut unknown_skipped_txs = 0;
         for (chained_source_txs, mpn_sourced_txs) in resps {
             for tx in chained_source_txs {
+                // ensure the transaction fee are greater than or equal to the minimum
+                // fee value the node is willing to accept
+                if let Some(tx_delta) = tx.tx_delta() {
+                    if tx_delta.tx.fee.lt(&ctx.min_fee) {
+                        log::debug!("skipping tx {:?}, fee too low", tx_delta);
+                        fee_skipped_txs += 1;
+                        continue;
+                    }
+                } else if let Some(mpn_tx) = tx.mpn_deposit() {
+                    if mpn_tx.payment.fee.lt(&ctx.min_fee) {
+                        log::debug!("skipping tx {:?}, fee too low", mpn_tx);
+                        fee_skipped_txs += 1;
+                        continue;
+                    }
+                } else {
+                    log::warn!("invalid tx {:?}", tx);
+                    unknown_skipped_txs += 1;
+                    continue;
+                }
                 ctx.mempool
                     .chain_sourced
                     .entry(tx)
                     .or_insert(TransactionStats { first_seen: now });
             }
             for tx in mpn_sourced_txs {
+                // ensure the transaction fee are greater than or equal to the minimum
```

Comment:
Let's forget about MPN transactions and skip them. These transactions are processed differently, let's only handle `ChainSourcedTx::TransactionAndDelta` transactions.